### PR TITLE
Fix various CLI problems including Windows fixes

### LINKED ISSUE #159: force "double dash" help on window
It now shows "/debug" in windows style: 

https://github.com/jessevdk/go-flags/blob/master/optstyle_windows.go

### LINKED ISSUE #283: Nerd job run command fails without input and with output
Command to reproduce:

    nerd job run --output /output overv/hugewrite

### PR DESCRIPTION
**Pull request by Overv**
The commandline syntax for inputs uses the `<JOB_DIR>:[DATASET_NAME]` syntax, but Windows paths usually have a colon in them, like `C:\something`. This currently makes it nigh impossible to use `nerd` on Windows.

Issues:

- [x] It requires a different format with `/flags` instead of `--flags`, which is confusing
- [x] It fails to parse `--output /output` because it thinks `/output` is an option instead of an argument
- [x] It checks if the job path is absolute using Windows requirements, which means that it does not consider `/input` absolute, for example
- [x] It requires absolute paths, which is inconvenient if they are quite long (also affects Linux/OSX) (@advanderveer why was it designed this way?)
- [x] It does not support `:` in paths, which is required for drive letters on Windows
- [x] Fix optional `--input` and `--output` errors
- [x] Refactor checks in `job run` and implement unit tests

Format can be fixed by adding `forceposix` build tag for the `go-flags` package:

https://github.com/jessevdk/go-flags/blob/4cc2832a6e6d1d3b815e2b9d544b2a4dfb3ce8fa/optstyle_windows.go

Update: Closes #159 
Closes #283

### CODE REVIEW

File: cmd/job_run.go [hunk 1]
```diff
@@ -35,6 +35,12 @@ func JobRunFactory(ui cli.Ui) cli.CommandFactory {
 	}
 }
 
+//isPathAbsUnix checks if a path is absolute on a Unix system.
+//Derived from https://github.com/golang/go/blob/1106512db54fc2736c7a9a67dd553fc9e1fca742/src/path/filepath/path_unix.go#L12
```
Comment (advanderveer) [inline cmd/job_run.go]: Move this to the svc package as it is a validation criteria for our implementation: check this out: https://godoc.org/gopkg.in/go-playground/validator.v9

File: cmd/job_run.go [hunk 2]
```diff
@@ -73,11 +79,17 @@ func (cmd *JobRun) Execute(args []string) (err error) {
 	vols := map[string]*svc.JobVolume{}
 	for _, input := range cmd.Inputs {
 		parts := strings.Split(input, ":")
-		if len(parts) != 2 {
+		if len(parts) < 1 {
```
Comment (advanderveer) [inline cmd/job_run.go]: Maybe add a error case when there are more then 2 colons, this is probably always an error on the users side.

**Comment by advanderveer** [comment]
It would be cool if the --input= supports the tilde character (see go-homedir library), expands $HOME or other environment variables (shell probably does this). And resolves ./ or ../ using the go std libary filepath.clean(): https://golang.org/pkg/path/filepath/#Clean

File: cmd/job_run.go [hunk 3]
```diff
@@ -69,29 +70,45 @@ func (cmd *JobRun) Execute(args []string) (err error) {
...
-	var inputDataset string
+	var inputDatasets []string
 	vols := map[string]*svc.JobVolume{}
 	for _, input := range cmd.Inputs {
+		var inputDataset string
+
 		parts := strings.Split(input, ":")
-		if len(parts) != 2 {
+
+		//Two accepted cases:
```
Comment (advanderveer) [inline cmd/job_run.go]: ❤️ the docs!

File: cmd/job_run.go [hunk 4]
```diff
@@ -69,29 +70,45 @@ func (cmd *JobRun) Execute(args []string) (err error) {
 
 	//start with input volumes
 	//@TODO move this logic to a separate package and test is
-	var inputDataset string
+	var inputDatasets []string
 	vols := map[string]*svc.JobVolume{}
 	for _, input := range cmd.Inputs {
```
Comment (advanderveer) [inline cmd/job_run.go]: Could you take a look at the possibility of moving these path "parsing" and assertion logic to a separate function and add unit tests? Seems perfect for table testing: [https://blog.golang.org/subtests](https://blog.golang.org/subtests)!

**Comment by Overv** [comment]
Input path parsing and validation is now unit tested.

**Pull request merged**